lib/tree.rb

```diff
@@ -1,38 +1,77 @@
 class NoApplesError < StandardError; end
 
+
+
 class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_reader :height, :age, :apples, :alive
 
-  def initialize
+  def initialize(height, age, apples, alive)
+    @height = height
+    @age = age
+    @apples = apples
+    @alive = alive
   end
 
   def age!
+    @age = @age + 1
+    @height = @height + Random.new.rand(0.5..2.5)
+    self.add_apples
   end
 
   def add_apples
+    rn = Random.new
```

Comment:
You don't have to initialize a new generator to use rand. It's an alias for Random::DEFAULT.rand so you can get away with just writing rand(1..10) here.